jetbrains-core/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryService.kt

```diff
@@ -48,57 +60,23 @@ interface TelemetryService : Disposable {
         return metricEvent
     }
 
-    suspend fun sendFeedback(sentiment: Sentiment, comment: String)
-
-    companion object {
-        @JvmStatic
-        fun getInstance(): TelemetryService = ServiceManager.getService(TelemetryService::class.java)
-
-        @JvmStatic
-        fun syncPublisher() = ApplicationManager.getApplication().messageBus.syncPublisher(TELEMETRY_TOPIC)
-
-        @JvmStatic
-        fun subscribe(notifier: TelemetryEnabledChangedNotifier) {
-            ApplicationManager.getApplication().messageBus.connect().subscribe(TELEMETRY_TOPIC, notifier)
-        }
-
-        private val TELEMETRY_TOPIC: Topic<TelemetryEnabledChangedNotifier> = Topic.create(
-            "TELEMETRY_ENABLED_TOPIC",
-            TelemetryEnabledChangedNotifier::class.java
-        )
-    }
-}
-
-interface TelemetryEnabledChangedNotifier {
-    fun notify(isTelemetryEnabled: Boolean)
-}
+    fun record(metricEventMetadata: MetricEventMetadata, buildEvent: MetricEvent.Builder.() -> Unit): MetricEvent {
+        val builder = DefaultMetricEvent.builder()
+        builder.awsAccount(metricEventMetadata.awsAccount)
+        builder.awsRegion(metricEventMetadata.awsRegion)
 
-class DefaultTelemetryService(settings: AwsSettings) :
-    TelemetryService, TelemetryEnabledChangedNotifier {
-    private val publisher = DefaultTelemetryPublisher()
-    var batcher: TelemetryBatcher = DefaultTelemetryBatcher(publisher)
-        set(value) {
-            batcher.setBatcher(value)
-            field = value
-        }
+        buildEvent(builder)
```

Comment:
Why did this move down after attaching account ID? Does it change anything?

Reply:
No it doesn't, just thought it read easier

Reply:
Cool, the diff was just hard to read here so I couldn't tell if there was anything weird with the change

Reply:
i have moved it back to its original position, it now does have some changes though :D